karate-core/src/main/java/com/intuit/karate/core/ScenarioRuntime.java

```diff
@@ -484,4 +502,23 @@ public String toString() {
         return scenario.toString();
     }
 
+    public void evaluateScenarioName() {
+        String scenarioName = this.scenario.getName();
+        boolean wrappedByBackTick = scenarioName != null && scenarioName.length() > 1 && '`' == scenarioName.charAt(0) && '`' == scenarioName.charAt((scenarioName.length() - 1));
+        if (wrappedByBackTick) {
+            String evaluatedScenarioName = this.engine.evalJs(scenarioName).getAsString();
+            this.scenario.setName(evaluatedScenarioName);
+        }
+    }
+
+    public void evaluateScenarioDescription() {
```

Comment:
* is fixing the description really necessary (then I have to include this in the JSON when we serialize a test result)
* is supporting the `\` to spill over 2 lines really necessary

I'd like to keep it simple :|

Reply:
what I meant when I said we need to care about multiple lines is I believe the first line always goes into the `name` and any extra lines go into the `description`. if that is the case, I propose we only interpolate the `name`

for those who want extra / dynamic docs - the new `doc` keyword is what I'd like users to use, and this can go into any `Step`

Reply:
when I thought about 2 lines I was thinking about wrapping lines - different developers might have different views in terms of the size of a line / line wrap. I agree that anything past 2 lines is probably bad practice (and reports won't look pretty at all) but what do you think about still supporting any lines and leaving the "good practice" of keeping the name short and simple to the test scripters?

I'm just thinking that might need to throw an exception if we want to limit to 2 lines and one day someone will want "just" the 3 lines  :)

The description was already in the code (saw it in the replace). Are you planning on removing it all together? I agree with not evaluating as the scenario outline should always do the same validations so a static description should suffice.

Reply:
I hear ya. I'd be okay if this wasn't cucumber, the syntax is biased towards "one line at a time", the only way to span multiple lines is to use the triple-quotes `"""`

so I'd really like to stick to one-line for the name. I also made this commit to verify what I was saying, that the parser already gives you the name and description separate: https://github.com/intuit/karate/commit/77310acf8ff0fe569d67419e909626d6778d4dd7

not planning to remove the description but we have now created a JSON for representing "an executed scenario" - see the `toKarateJson()` methods in `ScenarioResult` etc. right now I'm not including the scenario description because it can always be derived from the original feature-file.

Reply:
so yes, good point - if the description is already being included in the replace - we can decide to not do it anymore. and one reason is we have a new way of adding fancy descriptions in-line. the name is important to stand out in reports

Reply:
got it - will remove the usage of \ altogher

Reply:
will remove the option of using string interpolation in the description but for now will leave the replace on the description - can probably be marked as deprecated or something but might be important to keep it for backwards compability